### PR TITLE
feat: PromQL binary expr planner

### PR DESCRIPTION
**Pull request by waynexia**
Signed-off-by: Ruihang Xia <waynestxia@gmail.com>

I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Plan PromQL `Binary` expression to datafusion `LogicalPlan`. Plan and expression are mixed in PromQL, so translating this part might not be very straightforward:
- Literals should be processed separately because they are not plan in DF. 
- Binary ops on columns are translated to an `INNER JOIN ON TIME INDEX`, with operator applied to each value-column.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/promql/src/planner.rs [hunk 1]
```diff
@@ -139,14 +141,56 @@ impl<S: ContextProvider> PromPlanner<S> {
...
-
-                UnsupportedExprSnafu {
-                    name: "Prom Binary Expr",
+            PromExpr::Binary(PromBinaryExpr { lhs, rhs, op, .. }) => {
+                match (
+                    Self::try_build_literal_expr(lhs),
+                    Self::try_build_literal_expr(rhs),
+                ) {
+                    (Some(_lhs), Some(_rhs)) => {
+                        todo!("is this a legal case?")
```
Comment (evenyag) [inline src/promql/src/planner.rs:150]: Is this a binary expr between two literals? Looks like promql supports this

https://github.com/prometheus/prometheus/blob/6dcfb717403e4a58c770dad796a09e9570a7a684/promql/testdata/literals.test#L10

Comment (killme2008) [inline src/promql/src/planner.rs:150]: I think it's a legal case; you can execute `1+1` in Prometheus too.

Comment (waynexia) [reply src/promql/src/planner.rs:150]: I don't mean the expression itself. I'm considering if this case should be captured before this match. As the literal-only expression is not a plan. I'll change this to a todo comment and return an unsupported error for now.

Comment (waynexia) [reply src/promql/src/planner.rs:150]: https://github.com/GreptimeTeam/greptimedb/pull/889#discussion_r1089860423

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by yuanbohan — APPROVED** [review]

**Pull request merged**